### PR TITLE
gui: Don't set default path editing existing folders without label (fixes #4297)

### LINKED ISSUE #4297: Default path set when editing existing folder without label
Discovered in:
https://forum.syncthing.net/t/data-path-vanished-error-box-appearing-folder-stops-sync-ing-megabug/10340/5

To reproduce, just edit any folder without a label and the default path will appear instead of the previous one.

### PR DESCRIPTION
**Pull request by imsodin**
I put the condition about  whether the path should be changed in the watch into a separate function to make them consistent. This is not yet tested, that's up next.

### CODE REVIEW

File: gui/default/syncthing/core/syncthingController.js [hunk 1]
```diff
@@ -1388,14 +1392,14 @@ angular.module('syncthing.core')
         });
 
         $scope.$watch('currentFolder.label', function (newvalue) {
-            if (!$scope.config.options || !$scope.config.options.defaultFolderPath || $scope.editingExisting || !$scope.folderEditor.folderPath.$pristine || !newvalue) {
+            if (!newvalue || shouldNotSetDefaultPath) {
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Call the function?

**Comment by calmh** [comment]
I'd suggest making it `shouldSetDefaultFolderPath` and inverting the all the `!... || !... || !...` to `... && ... && ...` instead, moving the `!` to the call site.

**Comment by calmh** [comment]
@st-review lgtm appears to work in my testing

**Comment by st-review** [comment]
@calmh: Noted! Need another LGTM or explicit merge command.

**Comment by imsodin** [comment]
I can now confirm that this specific problem is fixed and adding a default path still works when adding a new folder

**Comment by calmh** [comment]
Meh slowpokes

**Comment by calmh** [comment]
@st-review merge it, what could possibly go wrong

**Comment by st-review** [comment]
:ok_hand: Merged as 77578e8aac65c0660d98ad7c4f65b020718cb7c2. Thanks, @imsodin!

**Comment by calmh** [comment]
@imsodin thanks for quickly identifying, filing, and fixing! ❤️

**Comment by imsodin** [comment]
@calmh Thanks for highlighting the good parts about this xD

**Pull request closed without merging**